sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -139,6 +227,10 @@ private ISpan createChild(
 
     Objects.requireNonNull(parentSpanId, "parentSpanId is required");
     Objects.requireNonNull(operation, "operation is required");
+    if (idleTimeout != null) {
+      // reset the timer when a new child gets added to extend the idle transaction
```

Comment:
here, perhaps we'd need to change status to `FinishStatus.notFinished()`, in case a new child was added AFTER the idle timeout, but we are still waiting for another child to finish. This is needed if we decide to reset the timer, no matter what, even if the status is already Finishing

Reply:
We've decided to reset the timer when the child has finished, rather than when it was created, so this avoids the problem entirely